finality-aleph/src/environment.rs

```diff
@@ -247,55 +226,67 @@ where
     // Pulls requests from the priority queue (sorted by scheduled time) and sends them to random peers
     // as long as they are scheduled at time <= curr_time
     pub(crate) fn trigger_tasks(&mut self) {
-        loop {
+        while let Some(request) = self.requests.peek() {
             let curr_time = time::Instant::now();
-            if self.requests.is_empty() || self.requests.peek().unwrap().scheduled_time > curr_time
-            {
+            if request.scheduled_time > curr_time {
                 break;
             }
-            let task = self.requests.pop().expect("Queue must be non-empty").task;
+            let request = self.requests.pop().expect("The element was peeked");
+            let task = request.task;
+
             match task {
                 Task::CoordRequest(coord) => {
-                    debug!(target: "env", "Starting request for {:?}", coord);
-                    // If we already have a unit with such a coord in our store then there is no need to request it.
-                    // It will be sent to consensus soon (or have already been sent).
-                    if !self.store.contains_coord(&coord) {
-                        let message =
-                            self.form_network_message(ConsensusMessage::FetchRequest(coord));
-                        let command = NetworkCommand::SendToRandPeer(message);
-                        self.place_network_command(command);
-                        debug!(target: "env", "Fetch request for {:?} sent.", coord);
-                        self.requests.push(ScheduledTask::new(
-                            Task::CoordRequest(coord),
-                            curr_time + FETCH_INTERVAL,
-                        ));
-                    } else {
-                        debug!(target: "env", "Request dropped as the unit is in store already {:?}", coord);
-                    }
```

Comment:
The method `trigger_tasks` was quite heave, so I delegated handling the two types of tasks to separate methods